gammapy/obs/tests/test_observation.py

```diff
@@ -9,3 +13,97 @@
 
 def test_ObservationTable():
     ObservationTable()
+
+def test_filter_observations():
+    # create random observation table
+    observatory_name='HESS'
+    n_obs = 10
+    obs_table = make_test_observation_table(observatory_name, n_obs)
+
+    # test no selection: input and output tables should be the same
+    filtered_obs_table = obs_table.filter_observations()
+    assert len(filtered_obs_table) == len(obs_table)
+
+    # filter some pars and check the correspoding values in the columns
+
+    # test box selection in obs_id
+    variable = 'OBS_ID'
+    value_min = 2
+    value_max = 5
+    selection = dict(shape='box', variable=variable,
+                     value_min=value_min, value_max=value_max)
+    filtered_obs_table = obs_table.filter_observations(selection)
+    assert (value_min < filtered_obs_table[variable]).all()
+    assert (filtered_obs_table[variable] < value_max).all()
+
+    # test box selection in obs_id inverted
+    selection = dict(shape='box', variable=variable,
+                     value_min=value_min, value_max=value_max, inverted=True)
+    filtered_obs_table = obs_table.filter_observations(selection)
+    selection = dict(shape='box', variable=variable,
+                     value_min=value_min, value_max=value_max, inverted=True)
+    filtered_obs_table = obs_table.filter_observations(selection)
+
+    # test circle selection in obs_id
+    variable = 'OBS_ID'
+    center = 4
+    radius = 2
+    selection = dict(shape='circle', variable=variable,
+                     center=center, radius=radius)
+    filtered_obs_table = obs_table.filter_observations(selection)
+    assert (center - radius < filtered_obs_table[variable]).all()
+    assert (filtered_obs_table[variable] < center + radius).all()
+
+    # test box selection in alt
+    variable = 'ALT'
+    value_min = Angle(Angle(70., 'degree').radian, 'radian')
```

Comment:
Use the `to` method ... it's a bit simpler and does the same thing (give you an `Angle` object back):
```python
Angle(42, 'deg').to('radian')
```